### ISSUE TITLE
Truncated 0 from float fields in facet results

### ISSUE THREAD

**Issue by karanhudia** [typesense/typesense#1432]
## Description

I am trying to query float type facets, and the results have all the leading `0`'s truncated. However, it returns correct value in the typesense document and also in the `filter_by` it takes the correct value as input to filter the results.

Only facet values are affected.  

## Steps to reproduce

The search JSON-
```
{
  "q": "*",
  "group_by": "product_id",
  "include_fields": "structured_specifications_float_weight_limit",
  "query_by": "title_en",
  "filter_by": "structured_specifications_float_weight_limit :> 0",
  "facet_by": "structured_specifications_float_weight_limit"
}
```
1. Facet results 
<img width="619" alt="Screenshot 2023-12-18 at 12 02 52 PM" src="https://github.com/typesense/typesense/assets/12070443/5cb67406-d8fd-481f-8bc5-08301744d24f">

2. Same facet field in document
<img width="572" alt="Screenshot 2023-12-18 at 12 06 16 PM" src="https://github.com/typesense/typesense/assets/12070443/3348602b-edbc-4f02-9b8b-8ff790a4110c">

## Expected Behavior

The expected value should have all the zeroes as the original documents, for example in this case, values should be `100`, `120`, `150`.

## Actual Behavior

The actual value have all the zeroes truncated in the facet results, for example in this case, values are `1`, `12`, `15`.

## Metadata

**Typesense Version**: `0.25.1`

**OS**: Self hosted Kubernetes pod

**Comment by jasonbosco** [comment]
@karanhudia We have a potential fix for this issue in `0.25.2.rc15`. Could you upgrade to it and let me know if it looks good now?

**Comment by karanhudia** [comment]
> @karanhudia We have a potential fix for this issue in `0.25.2.rc15`. Could you upgrade to it and let me know if it looks good now?

@jasonbosco Wow! that does fix it. Thanks for the quick support 🙌

**Comment by jasonbosco** [comment]
Awesome, thank _you_ for confirming and catching this issue!
